scripts/anvil_tools/set_up_anvil_workspaces.py

```diff
@@ -0,0 +1,309 @@
+    successes = report_df.workspace_setup_status.str.count("Success").sum()
+    fails = report_df.workspace_setup_status.str.count("Failed").sum()
+    total = successes + fails
+    print(f"Number of workspaces passed set-up: {successes}/{total}")
+    print(f"Number of workspaces failed set-up: {fails}/{total}")
+    print(f"All workspace set-up (success or fail) details available in output file: {output_filename}")
+
+
+def add_members_to_workspace(workspace_name, auth_domain_name, project="anvil_datastorage"):
+    """Add members to workspace permissions."""
+
+    acls = []
+    # add auth domain as READER, anvil-admins as OWNER
+    acls.append({'email': f'{auth_domain_name}@firecloud.org', 'accessLevel': 'READER', 'canShare': False, 'canCompute': False})
+    acls.append({'email': 'anvil-admins@firecloud.org', 'accessLevel': 'OWNER', 'canShare': True, 'canCompute': True})
+
+    json_request = json.dumps(acls)
+
+    # request URL for updateWorkspaceACL
+    uri = f"https://api.firecloud.org/api/workspaces/{project}/{workspace_name}/acl?inviteUsersNotFound=false"
+
+    # Get access token and and add to headers for requests.
+    headers = {"Authorization": "Bearer " + get_access_token(), "accept": "*/*", "Content-Type": "application/json"}
+    # -H  "accept: */*" -H  "Authorization: Bearer [token] -H "Content-Type: application/json"
+
+    # capture response from API and parse out status code
+    response = requests.patch(uri, headers=headers, data=json_request)
+    status_code = response.status_code
+
+    emails = [acl['email'] for acl in acls]
+    # print success or fail message based on status code
+    if status_code != 200:
+        print(f"WARNING: Failed to update {project}/{workspace_name} with the following user(s)/group(s): {emails}.")
+        print("Check output file for error details.")
+        return status_code, response.json()
+    else:
+        print(f"Successfully updated {project}/{workspace_name} with the following user(s)/group(s): {emails}.")
+        return status_code, emails
+
+
+def check_workspace_exists(workspace_name, project="anvil-datastorage"):
+    """Determine if a workspace of given namespace/name already exists."""
+
+    # don't need full response - could be very large and time consuming
+    uri = f"https://api.firecloud.org/api/workspaces/{project}/{workspace_name}?fields=owners,workspace.createdBy,workspace.authorizationDomain"
+
+    # Get access token and and add to headers for requests.
+    # -H  "accept: application/json" -H  "Authorization: Bearer [token]
+    headers = {"Authorization": "Bearer " + get_access_token(), "accept": "application/json"}
+
+    # capture response from API and parse out status code
+    response = requests.get(uri, headers=headers)
+    status_code = response.status_code
+
+    if status_code == 404:          # if workspace does not exist
+        return False, None, None
+    elif status_code == 200:        # if workspace already exists
```

Comment:
here you can just do `if status_code == 200:` and 2 lines down you can delete the else